tls/s2n_config.c

```diff
@@ -761,8 +761,8 @@ int s2n_config_add_ticket_crypto_key(struct s2n_config *config,
     uint32_t ticket_keys_len = 0;
     POSIX_GUARD_RESULT(s2n_set_len(config->ticket_keys, &ticket_keys_len));
     S2N_ERROR_IF(ticket_keys_len >= S2N_MAX_TICKET_KEYS, S2N_ERR_TICKET_KEY_LIMIT);
-
     S2N_ERROR_IF(name_len == 0 || name_len > S2N_TICKET_KEY_NAME_LEN || s2n_find_ticket_key(config, name), S2N_ERR_INVALID_TICKET_KEY_NAME_OR_NAME_LENGTH);
+    S2N_ERROR_IF(name_len == key_len && memcmp(name, key, key_len) == 0, S2N_ERR_INVALID_TICKET_KEY_NAME_OR_NAME_LENGTH);
```

Comment:
Consider using `POSIX_ENSURE` instead of `S2N_ERROR_IF` as we are moving away from the older macros. See issue: https://github.com/aws/s2n-tls/issues/2635